.circleci/config.yml

```diff
@@ -246,9 +258,104 @@ jobs:
+            git clone https://github.com/netsec-ethz/scionlab.git -b master ./
+
+      - run:
+          name: Extract host configuration info from testdata.yaml fixture
+          working_directory: ~/previous_version
+          command: |
+            pip install pyyaml
+            python .circleci/setup/generate-host-envs.py
+
+      - run:
+          name: Build container images
+          working_directory: ~/previous_version/.circleci
+          command: |
+            #docker build --file ~/project/previous_version/deploy/Dockerfile-django --tag scionlab_django ~/project/ # We are only extending the production image
+            sed -i '/.circleci/d' ~/previous_version/.dockerignore  # The .dockerignore is for production, we'll need this
+            docker-compose build
+
+
+      - run:
+          name: Start old coordinator
+          working_directory: ~/previous_version/.circleci
+          <<: *start_coordinator
+
+      - run:
+          name: Start ASes running old version in containers
+          working_directory: ~/previous_version/.circleci
+          command: |
+            set -x
+            docker-compose up -d as1301 as1303 as1305
+
+            for c in as1301 as1303 as1305; do
+              docker-compose exec $c /bin/bash -c \
+                'apt-get install --assume-yes -t all scionlab; \
+                 scionlab-config --host-id ${SCIONLAB_HOST_ID} --host-secret ${SCIONLAB_HOST_SECRET} --url http://coord:8000'
+            done
+
+            docker-compose logs
+
+      - run:
+          name: Check AS services status running old config
+          working_directory: ~/previous_version/.circleci
+          <<: *check_services_status
+
+      # Starting upgrade integration tests
+
+      - run:
+          name: Rebuild and start coordinator from current branch
+          working_directory: ~/project/.circleci
+          command: |
+            sed -i '/.circleci/d' ~/project/.dockerignore  # The .dockerignore is for production, we'll need this
+            docker build --file ~/project/deploy/Dockerfile-django --tag scionlab_django ~/project/
+            chown circleci:circleci -R ~/previous_version/run/
+            cp -r ~/previous_version/run ~/project/
+            docker-compose up -d --force-recreate --build coord
+
+      - run:
+          name: Run client cronjob
+          working_directory: ~/previous_version/.circleci
+          command: |
+            set -x
+            # Simulate command that is executed on hosts with upgrade cronjob (VMs), upgrade package
+            for c in as1301 as1303 as1305; do
+              docker-compose exec -T $c /bin/bash -c \
+                'echo "deb [trusted=yes] https://packages-test.netsec.inf.ethz.ch/debian all main" | sudo tee /etc/apt/sources.list.d/scionlab.list; \
+                 apt-get update; rm /var/lib/scion/*; DEBIAN_FRONTEND=noninteractive apt-get install -y --only-upgrade -t all scionlab'
```

Comment:
I think the `-t` is not necessary. Or what does it do?

Reply:
The `rm /var/lib/scion/*`  should happen in the package install script, right? Or do the cron jobs do this already?

Reply:
>I think the -t is not necessary. Or what does it do?

>>docker-compose exec: -T                Disable pseudo-tty allocation.

Cron jobs don't have a tty, right? Since this is supposed to simulate the commands as run from the cronjob, it disables the tty allocation.

>The rm /var/lib/scion/* should happen in the package install script, right?

Correct

>Or do the cron jobs do this already?

No

Reply:
Sorry, I meant the `-t` in  `apt-get install ... -t all`